src/next/middleware/http-auth.ts

```diff
@@ -20,10 +45,12 @@ export function handleHttpBasicAuth (
 			.toString()
 			.split(":");
 
-		if (givenUser === username && givenPassword === password)
-		{
-			return;
-		}
+		users.forEach((item) => {
+			if (givenUser === item.username && givenPassword === item.password)
+			{
+				return;
+			}
+		})
```

Comment:
That won't work, as you are just returning from the callback. You need to do it like this:

```js
const hasValidUser = users.some(item => givenUser === item.username && givenPassword === item.password);

if (hasValidUser)
{
    return;
}
```